packages/jellyfish-api-core/__tests__/category/account/withdrawFutureSwap.test.ts

```diff
@@ -0,0 +1,922 @@
+        expect(balance).toStrictEqual([])
+      }
+
+      {
+        // check tslaAddress
+        const balance = await testing.rpc.account.getAccount(tslaAddress)
+        expect(balance).toStrictEqual([`${swapAmount.toFixed(8)}@TSLA`])
+      }
+    }
+  })
+
+  it('should withdraw futureswap dusd to dtoken with utxo', async () => {
+    const swapAmount = 1
+    const tslaAddress = await testing.generateAddress()
+    await testing.rpc.account.accountToAccount(collateralAddress, { [tslaAddress]: `${swapAmount}@DUSD` })
+    await testing.generate(1)
+
+    const fswap: FutureSwap = {
+      address: tslaAddress,
+      amount: `${swapAmount.toFixed(8)}@DUSD`,
+      destination: 'TSLA'
+    }
+    await testing.rpc.account.futureSwap(fswap)
+    await testing.generate(1)
+
+    const utxo = await testing.container.fundAddress(tslaAddress, 50)
+
+    const withdrawAmount = swapAmount / 2
+    const withdrawFutureSwap: FutureSwap = {
+      address: tslaAddress,
+      amount: `${withdrawAmount.toFixed(8)}@DUSD`,
+      destination: 'TSLA'
+    }
+
+    const txid = await testing.rpc.account.withdrawFutureSwap(withdrawFutureSwap, [utxo])
+    await testing.generate(1)
+
+    const rawtx = await testing.container.call('getrawtransaction', [txid, true])
+    expect(rawtx.vin[0].txid).toStrictEqual(utxo.txid)
+    expect(rawtx.vin[0].vout).toStrictEqual(utxo.vout)
+
+    // check the future is in effect
+    {
+      const pendingFutures = await testing.container.call('listpendingfutureswaps')
+      expect(pendingFutures.length).toStrictEqual(1)
+      expect(pendingFutures[0].owner).toStrictEqual(tslaAddress)
+      expect(pendingFutures[0].source).toStrictEqual(`${(swapAmount - withdrawAmount).toFixed(8)}@DUSD`)
+      expect(pendingFutures[0].destination).toStrictEqual('TSLA')
+
+      // check live/economy/dfip2203_*
+      const attributes = await testing.rpc.masternode.getGov(attributeKey)
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_current']).toStrictEqual([`${(swapAmount - withdrawAmount).toFixed(8)}@DUSD`])
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_burned']).toBeUndefined()
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_minted']).toBeUndefined()
+
+      // dfip2203 burn should be empty
+      const burnBefore = await testing.rpc.account.getBurnInfo()
+      expect(burnBefore.dfip2203).toStrictEqual([])
+
+      {
+        // check contractAddress
+        const balance = await testing.rpc.account.getAccount(contractAddress)
+        expect(balance).toStrictEqual([`${(swapAmount - withdrawAmount).toFixed(8)}@DUSD`])
+      }
+
+      {
+        // check tslaAddress
+        const balance = await testing.rpc.account.getAccount(tslaAddress)
+        expect(balance).toStrictEqual([`${withdrawAmount.toFixed(8)}@DUSD`])
+      }
+    }
+  })
+
+  it('should not withdraw invalid futureswap dtoken to dusd - before settle block', async () => {
```

Comment:
For invalid parameter values, I think you can choose either
dtoken to dusd or dusd to dtoken but not both

Because you are testing input value for the function, which should be the same for both scenarios

Reply:
yeah, can have a single `it()` to avoid duplicate tests. But if you have extra in one case, you can add them.

Reply:
Keeping both, as there is an extra case